rust/Makefile

```diff
@@ -11,6 +11,8 @@ extra-$(CONFIG_RUST) += exports_alloc_generated.h exports_kernel_generated.h
 
 ifndef CONFIG_RUST_BUILD_ASSERT_DENY
 obj-$(CONFIG_RUST) += build_error.o
+else
+extra-$(CONFIG_RUST) += build_error.o
 endif
```

Comment:
Since we have both arms now, inverting the condition to use `ifdef` is maybe more readable.